### PR TITLE
Sd 256 whitelist address per issuer

### PR DESCRIPTION
**Pull request by githinho**

### CODE REVIEW

**Review by josipkoncurat — APPROVED** [review]
Looks good!

File: src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt [hunk 1]
```diff
@@ -4,40 +4,47 @@ import com.ampnet.identityservice.config.ApplicationProperties
...
-            .sendSafely()?.transactionCount ?: return null
-        val gasPrice = web3j.ethGasPrice().sendSafely()?.gasPrice ?: return null
-        val transaction = Transaction
-            .createContractTransaction(credentials.address, nonce, gasPrice, approveWalletCall)
-
-        val transactionHash = web3j.ethSendTransaction(transaction).sendSafely()?.transactionHash
-        logger.info { "Sent transaction: $transactionHash" }
-        return transactionHash
+        val nonce = web3j.ethGetTransactionCount(credentials.address, DefaultBlockParameterName.LATEST)
+            .send().transactionCount ?: return null
```
Comment (josipkoncurat) [inline src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt:36]: replace send() with sendSafely()

File: src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt [hunk 2]
```diff
@@ -4,40 +4,47 @@ import com.ampnet.identityservice.config.ApplicationProperties
...
+        val gasPrice = web3j.ethGasPrice().send().gasPrice ?: return null
```
Comment (josipkoncurat) [inline src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt:37]: replace send() with sendSafely()

File: src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt [hunk 3]
```diff
@@ -4,40 +4,47 @@ import com.ampnet.identityservice.config.ApplicationProperties
...
+        val function = Function("approveWallet", listOf(issuerAddress.toAddress(), address.toAddress()), emptyList())
```
Comment (josipkoncurat) [inline src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt:38]: you can replace string with IIssuer.FUNC_APPROVEWALLET

File: src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt [hunk 4]
```diff
@@ -4,40 +4,47 @@ import com.ampnet.identityservice.config.ApplicationProperties
...
+        val encoded = FunctionEncoder.encode(function)
+        val chainId = 80001L
```
Comment (josipkoncurat) [inline src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt:40]: Maybe just a note that this refers to the: Matic Mumbai Testnet chainId

**Review by josipkoncurat — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt (hunk 3 above)
Comment (githinho) [reply src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt:38]: This is not call on Issuer contract.

File: src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt (hunk 4 above)
Comment (githinho) [reply src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt:40]: This will be changed in task SD-257 to support multiple chains.

**Review by filipdujmusic — APPROVED** [review]

**Pull request merged**